flutter/lib/src/sentry_flutter.dart

```diff
@@ -9,51 +9,49 @@ import 'sentry_flutter_options.dart';
 
 import 'default_integrations.dart';
 import 'file_system_transport.dart';
-import 'sentry_flutter_options.dart';
 import 'version.dart';
 // conditional import for the iOSPlatformChecker
 // in browser, the iOSPlatformChecker will always return false
 // the iOSPlatformChecker is used to run the loadContextsIntegration only on iOS.
 // this injected PlatformChecker allows to test this behavior
 import 'web_platform_checker.dart' if (dart.library.io) 'platform_checker.dart';
 
+/// Configuration options callback
+typedef FlutterOptionsConfiguration = FutureOr<void> Function(
+    SentryFlutterOptions);
+
 /// Sentry Flutter SDK main entry point
 mixin SentryFlutter {
   static const _channel = MethodChannel('sentry_flutter');
 
   static Future<void> init(
-    OptionsConfiguration optionsConfiguration, {
+    FlutterOptionsConfiguration optionsConfiguration, {
```

Comment:
The user should be able to set Flutter specific configuration 😅

Reply:
mm but this is supposed to work ootb with inheritance, right?

other option would be generics:

typedef FlutterOptionsConfiguration = FutureOr<void> Function<T extends SentryOptions>(
    T);

not sure about the syntax here, but something similar to it.

Reply:
No, unfortunately not.

Reply:
if we need to cast it's ok, but if the Client needs to `OptionsConfiguration<T>` is suboptimal, ideally, this is just a lambda, no boilerplate.

I still fail to understand why we need a `FlutterOptionsConfiguration` though, our sample uses the standard `OptionsConfiguration` and still works well with the `SentryFlutterOptions`, so not sure if I am missing something.

Other than that, I guess this PR is pretty good and ready to be landed, so would you mind elaborating on why do we need a specific `FlutterOptionsConfiguration`?

Reply:
ok I guess I got it, our samples were not using any method only visible to the `SentryFlutterOptions`